src/loam/allocation.rs

```diff
@@ -181,13 +181,13 @@ impl Allocator {
             .unwrap()
     }
 
-    pub fn unhash6(&mut self, digest: &Wide) -> [Wide; 6] {
-        let mut preimage = [Wide::widen(LE::from_canonical_u32(0)); 6];
+    pub fn unhash5(&mut self, digest: &Wide) -> [Wide; 5] {
+        let mut preimage = [Wide::widen(LE::from_canonical_u32(0)); 5];
 
         self.preimage_cache
             .get(digest)
             .map(|digest| {
-                preimage.copy_from_slice(&digest[..6]);
+                preimage.copy_from_slice(&digest[..5]);
```

Comment:
nit: This `digest` parameter is misnamed and is actually the preimage (this is also applicable to the other unhashX functions in this file (cc @winston-h-zhang)